src/main/java/lotto/domain/LottoResult.java

```diff
@@ -0,0 +1,47 @@
+package lotto.domain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoResult {
+
+    private static final int LOTTO_NUMBERS_LENGTH = 6;
+    private Map<Integer, Integer> lottoResultMap;
+
+    public LottoResult(LottoTicket lottoTicket, LottoNumbers winningNumbers) {
+        initializeLottoResultMap();
+        countResults(lottoTicket, winningNumbers);
+    }
+
+    private void initializeLottoResultMap() {
+        lottoResultMap = new HashMap<>();
+        for (int i = 0; i <= LOTTO_NUMBERS_LENGTH; i++) {
+            lottoResultMap.put(i, 0);
+        }
+    }
+
+    private void countResults(LottoTicket lottoTicket, LottoNumbers winningNumbers) {
+        List<LottoNumbers> allLottoNumbers = lottoTicket.allLottoNumbers();
```

Comment:
```suggestion
        LottoResult result = lottoTicket.match(winningNumbers);
```
이와 같은 구조로 구현하는 것은 어떨까?
LottoResult는 지금 코드와 같이 Map<Integer, Integer> lottoResultMap에 일급 콜렉션이다.